### DISCUSSION TITLE
Variable font axes controlled by MIDI

### DISCUSSION THREAD

**Discussion by mgpjm** [coldtype/coldtype#96] [Q&A] [answered]
Hi Rob,

just wondering, can you publicate a short tutorial about this video somewhere? https://www.instagram.com/p/CFlHysxBhAq/
Or, at least, release the code, is it really hard to get it from the blurry video :) (Or is already published and I missed it?)
Hope the community can enjoy it

Thank you!
m

**Comment by stenson** [comment, marked as answer]
Hopefully I'll be able to do a tutorial about that sometime soon, but there is some general info about using a MIDI controller with coldtype here: https://coldtype.goodhertz.com/tutorials/midi.html, and here's some more specific code for controlling a variable font with a controller:

```python
from coldtype import *
from coldtype.midi.controllers import LaunchControlXL

@animation((1080, 1080), bg=1, rstate=1)
def render(f, rstate):
    nxl = LaunchControlXL(rstate.midi)
    
    return P(
        (P(f.a.r).f(Gradient.V(f.a.r,
            hsl(nxl(20, 0.25), a=0.5),
            hsl(nxl(10, 0.45), a=0.5)))),
        (StSt("MIDI", Font.MutatorSans()
            , fontSize=20+nxl(12, 0.25)*500
            , wdth=nxl(11, 0.25)
            , wght=nxl(21)
            , tu=-250+nxl(22)*500
            , r=1
            , ro=1)
            .align(f.a.r, tv=1)
            .f(1)))
```

The trick here is that I'm using a LaunchControlXL as my controller, so there are some utility functions in Coldtype to make working with that controller a little easier, but that general documentation should have some tips and tricks on how to use any controller.

**Reply by mgpjm** [reply]
Dear Rob, 
it definitely works! As an Ableton user I have a Launchkey mini, but still is definitely doing its job.

I'm happy to hear you will release a tutorial, even if this piece of code should really help me to config a live tool to play a little bit with fonts and music.

Thanks!
m
